cpp/tests/copying/detail_gather_tests.cu

```diff
@@ -30,18 +30,20 @@
 #include <cudf_test/table_utilities.hpp>
 #include <cudf_test/type_lists.hpp>
 
+#include <rmm/device_uvector.hpp>
+
 template <typename T>
 class GatherTest : public cudf::test::BaseFixture {
 };
 
 TYPED_TEST_CASE(GatherTest, cudf::test::NumericTypes);
 
 // This test exercises using different iterator types as gather map inputs
-// to cudf::detail::gather -- device_vector and raw pointers.
+// to cudf::detail::gather -- device_uvector and raw pointers.
 TYPED_TEST(GatherTest, GatherDetailDeviceVectorTest)
 {
   constexpr cudf::size_type source_size{1000};
-  rmm::device_vector<cudf::size_type> gather_map(source_size);
+  rmm::device_uvector<cudf::size_type> gather_map(source_size, rmm::cuda_stream_default);
```

Comment:
Just curious why are we adding stream support here? 
Is it because we are calling a detail API in this test?

Reply:
The stream is non optional for a `device_uvector` ctor.

Reply:
Yes, but then why use `device_uvector` here at all then. This is just test code after all.

Reply:
Because leaving them in makes my search in every PR much more manual and repetitive.

Reply:
And it doesn't hurt to make tests run faster and use less power!